File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--create-and-modify-custom-document-properties_1.cs
        void TestProperties()
        {
            Microsoft.Office.Core.DocumentProperties properties;
            properties = (Office.DocumentProperties)this.CustomDocumentProperties;

            if (ReadDocumentProperty("Project Name") != null)
            {
                properties["Project Name"].Delete();
            }

            properties.Add("Project Name", false,
                Microsoft.Office.Core.MsoDocProperties.msoPropertyTypeString,
                "White Papers");
        }

        private string ReadDocumentProperty(string propertyName)
        {
            Office.DocumentProperties properties;
            properties = (Office.DocumentProperties)this.CustomDocumentProperties;

            foreach (Office.DocumentProperty prop in properties)
            {
                if (prop.Name == propertyName)
                {
                    return prop.Value.ToString();
                }
            }
            return null;
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--create-and-modify-custom-document-properties_1.vb
    Sub TestProperties()
        Dim properties As Microsoft.Office.Core.DocumentProperties
        properties = CType(Me.CustomDocumentProperties, Office.DocumentProperties)

        If ReadDocumentProperty("Project Name") <> Nothing Then
            properties("Project Name").Delete()
        End If

        properties.Add("Project Name", False, _
            Microsoft.Office.Core.MsoDocProperties.msoPropertyTypeString, _
            "White Papers")
    End Sub

    Private Function ReadDocumentProperty(ByVal propertyName As String) As String
        Dim properties As Office.DocumentProperties
        properties = CType(Me.CustomDocumentProperties, Office.DocumentProperties)

        Dim prop As Office.DocumentProperty

        For Each prop In properties
            If prop.Name = propertyName Then
                Return prop.Value.ToString()
            End If
        Next

        Return Nothing
    End Function